### ISSUE TITLE
TypeScript: extraneous parens in BinaryExpression throw error

### ISSUE THREAD

**Issue by jwbay** [prettier/prettier#1903]
https://prettier.github.io/prettier/#%7B%22content%22%3A%22function%20foo()%20%7B%5Cn%20%20%20%20return%20a%20%26%26%20(b%20%26%26%20c)%3B%5Cn%7D%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Atrue%2C%22trailingComma%22%3A%22es5%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Atrue%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Atrue%2C%22doc%22%3Afalse%7D%7D

Simplified from RWC where it's a little less obvious 😅 

```ts
a && (b && c);
```
yields
```
TypeError: assert$2.strictEqual is not a function
```

**Comment by vjeux** [comment]
This is actually only an issue on the website, I forgot to mock strictEqual. The npm version is working fine. Thanks for the report, I'll fix it.

**Comment by vjeux** [comment]
I just published 1.4.1 which fixes it!
